core/src/main/java/org/wso2/msf4j/internal/MicroservicesRegistryImpl.java

```diff
@@ -92,8 +92,14 @@ public void addService(String basePath, Object service) {
     }
 
     public void removeService(Object service) {
-        services.remove(service);
-        updateMetadata();
+        Path path = service.getClass().getAnnotation(Path.class);
+        if (path != null) {
+            services.remove(path.value());
+            updateMetadata();
+        } else {
+            log.warn("Service removal failed. Microservice class '" + service.getClass().getName() +
+                     "' doesn't contain a root Path.");
```

Comment:
Better checking for if (path == null) and move corresponding logic to first block for better readability.